pkg/services/authn/clients/oauth.go

```diff
@@ -168,7 +168,7 @@ func (c *OAuth) Authenticate(ctx context.Context, r *authn.Request) (*authn.Iden
 
 	// This is required to implement OrgRole mapping for OAuth providers step by step
 	switch c.providerName {
-	case social.GenericOAuthProviderName, social.GitHubProviderName, social.GitlabProviderName, social.OktaProviderName:
+	case social.GenericOAuthProviderName, social.GitHubProviderName, social.GitlabProviderName, social.OktaProviderName, social.AzureADProviderName:
```

Comment:
I am pretty sure we can remove this entirely. But lets do that in a followup pr

Reply:
Yep, I'm planning to do that, because we need to align the Grafana.com integration a bit.